### PR TITLE
VDS-2159 Implement the edit company case

### PR DESCRIPTION
**Pull request by trueboroda**
Description: 

image

--

QA-test: VDS-2162

Demo-test:  VDS-2163

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-34-vds-2159-c57c14ed

### CODE REVIEW

File: src/app/graphql.module.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export function createApollo(
   const auth = setContext(async operation => {
     let token: string | null = null;
     switch (operation.operationName) {
+    case 'updateOrganization':
```
Comment (asvishnyakov) [inline src/app/graphql.module.ts]: Does we really need admin rights to update our own org?

File: src/app/modules/companies/components/company-edit/company-edit-form.model.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export const COMPANY_EDIT_FORM_INPUTS = {
+  name: new DynamicInputModel({
+    id: 'name',
+    label: 'Company Name',
+    validators: {
+      required: null,
+    },
+    errorMessages: {
+      required: 'The {{ label }} is required',
```
Comment (asvishnyakov) [inline src/app/modules/companies/components/company-edit/company-edit-form.model.ts]: We have constant for that

File: src/app/modules/companies/components/company-edit/company-edit.component.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  selector: 'vc-company-edit',
+
+  templateUrl: './company-edit.component.html',
+
+  styleUrls: [
+    './company-edit.component.scss',
+  ],
+
+})
```
Comment (asvishnyakov) [inline src/app/modules/companies/components/company-edit/company-edit.component.ts]: Remove unnecessary empty lines

File: src/app/modules/companies/components/company-edit/company-edit.component.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+export class CompanyEditComponent implements AfterViewInit, OnDestroy {
+  @ViewChild(DynamicNGBootstrapFormComponent, {
+    static: true,
+  })
+  formComponent!: DynamicNGBootstrapFormComponent;
+
+  selectedCompany$ = this.store.select(selectSelectedCompany);
```
Comment (asvishnyakov) [inline src/app/modules/companies/components/company-edit/company-edit.component.ts]: Move it down, after form fields, please

File: src/app/modules/companies/components/company-edit/company-edit.component.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  constructor(
+    private readonly formService: DynamicFormService,
+    private readonly store: Store,
+    private readonly route: ActivatedRoute
+  ) {  }
+
+  ngAfterViewInit(): void {
+    this.route.paramMap.pipe(takeUntil(this.unsubscriber)).subscribe(params => {
+      const id = params.get('id');
+      this.store.dispatch(getCompany({ id: id as string }));
```
Comment (asvishnyakov) [inline src/app/modules/companies/components/company-edit/company-edit.component.ts]: Refactor this code to support `/companies/current` url to use it as url in dashboard, please

File: src/app/modules/companies/store/companies.actions.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+export const getCompanyFailure = createAction(
+  '[Companies] Get Company Failure',
+  props<{ error: ApolloError }>()
+);
+
+export const setCompany = createAction(
+  '[Companies] Set Company',
+  props<{ data: PartialDeep<EditCompany> }>()
+);
```
Comment (asvishnyakov) [inline src/app/modules/companies/store/companies.actions.ts]: Looks like this action is unused and can be removed

File: src/app/modules/companies/store/companies.reducer.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  on(CompaniesActions.getCompanySuccess, (state, action): State  =>  {
+    const organization = action.data.organization;
+    return {
+      ...state,
+      selectedCompany: organization === null
+        ? null
+        : {
+          id: organization.id,
+          name: organization.name as string,
+        },
```
Comment (asvishnyakov) [inline src/app/modules/companies/store/companies.reducer.ts]: Similar code is repeated here 3 times. May be move it to (helper) function/class method?

File: src/app/modules/companies/store/companies.reducer.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      ...state.selectedCompany,
+      ...action.data,
+    },
+  })),
+  on(CompaniesActions.updateCompany, (state) : State => state),
+  on(CompaniesActions.updateCompanySuccess, (state, action): State  =>  {
+    const organization = action.data?.updateOrganization;
+    return {
+      ...state,
+      selectedCompany: !organization
```
Comment (asvishnyakov) [inline src/app/modules/companies/store/companies.reducer.ts]: Why `!` instead of `=== null` here?

File: src/app/store/index.ts [hunk 9]
```diff
@@ -1,35 +1,38 @@
...
-  createSelector,
-  MetaReducer
-} from '@ngrx/store';
-import { environment } from '../../environments/environment';
-import * as fromCountries from './countries/countries.reducer';
-import * as fromCompany from '../modules/registration/store/company.reducer';
-import * as fromCart from './cart/cart.reducer'
-import * as fromSectors from './sectors/sectors.reducer';
-import * as fromLogin from './login/login.reducer';
+import {
```
Comment (asvishnyakov) [inline src/app/store/index.ts:1]: Something wrong this changes in this file. May be line endings? May you check please?

File: src/app/store/index.ts [hunk 10]
```diff
@@ -1,35 +1,38 @@
...
+import * as fromCompanies from '../modules/companies/store/companies.reducer';
+
+export interface State {
+[fromCountries.countriesFeatureKey]: fromCountries.State;
+[fromCompany.companyRegistrationFeatureKey]: fromCompany.State;
+[fromCart.cartFeatureKey]: fromCart.State
+[fromSectors.sectorsFeatureKey]: fromSectors.State;
+[fromLogin.loginFeatureKey]: fromLogin.State;
+[fromCurrentCustomer.currentCustomerFeatureKey]: fromCurrentCustomer.State;
[fromCompanies.companiesFeatureKey]: fromCompanies.State;
```
Comment (asvishnyakov) [inline src/app/store/index.ts]: No new line. This file is usually autogenerated. Did you touched it?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/app/graphql.module.ts (hunk 1 above)
Comment (pushnitsa) [reply src/app/graphql.module.ts]: No, we should use current customer token
image

File: src/app/modules/companies/store/companies.reducer.ts [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import { EditCompany } from './../../../models/edit-company.model';
+import { PartialDeep } from 'type-fest';
+import { createReducer, on } from '@ngrx/store';
+import * as CompaniesActions from './companies.actions';
+
+export const companiesFeatureKey = 'companies';
+
+export interface State {
+  selectedCompany: PartialDeep<EditCompany> | null,
+  editCompany: PartialDeep<EditCompany> | null,
```
Comment (asvishnyakov) [inline src/app/modules/companies/store/companies.reducer.ts]: selectedCompany: PartialDeep<**EditCompany**> | null?

May be just Company then?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/app/modules/companies/store/companies.actions.ts (hunk 6 above)
Comment (trueboroda) [reply src/app/modules/companies/store/companies.actions.ts]: It is used. https://github.com/VirtoCommerce/vc-demo-xapi-app/blob/7a8cf4eb057f0bce9580b14916ac64c641421b7d/src/app/modules/companies/components/company-edit/company-edit.component.ts#L68

File: src/app/graphql.module.ts (hunk 1 above)
Comment (trueboroda) [reply src/app/graphql.module.ts]: In this case, we have AuthorizationError 
VirtoCommerce.ExperienceApiModule.Core.Infrastructure.Authorization.AuthorizationError: User doesn't have the required permission 'customer:update'

File: src/app/store/index.ts (hunk 10 above)
Comment (trueboroda) [reply src/app/store/index.ts]: Only the autogen was used. Possible I just have saved this file somewhen.

File: src/app/graphql.module.ts (hunk 1 above)
Comment (trueboroda) [reply src/app/graphql.module.ts]: Done. It temporally needs to use the xapi module from the branch VDS-2638.

File: src/app/modules/companies/store/companies.reducer.ts (hunk 11 above)
Comment (asvishnyakov) [reply src/app/modules/companies/store/companies.reducer.ts]: Not fixed

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**